src/services/mongoose/rewards.ts

```diff
@@ -0,0 +1,31 @@
+import { RewardAlreadyPresentError, UnknownRepositoryError } from "@domain/errors"
+import { User } from "./schema"
+
+// FIXME: improve boundary
+export const RewardsRepository = (accountId: AccountId) => {
+  const tentativelyAddNew = async (quizQuestionId: QuizQuestionId) => {
```

Comment:
we are not doing a rollback so maybe tentatively is not necessary

Reply:
the `tentative` is here to suggest that, if the rewards already exist, we will not credit it to the user. this may be misleading?